Paintroid/src/main/java/org/catrobat/paintroid/presenter/MainActivityPresenter.java

```diff
@@ -242,7 +243,12 @@ public void onNewImage() {
 
 	@Override
 	public void onNewImageFromCamera() {
-		interactor.createFile(this, CREATE_FILE_TAKE_PHOTO, null);
+		if (navigator.isSdkAboveOrEqualM() && !navigator.doIHavePermission(Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
```

Comment:
These copy-paste sections are starting to clutter up the presenter.
Please try to extract a common denominator (e.g. a function) to reduce repeating code snippets.